File: solutions/Valid Anagram/Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        if len(s) != len(t):
            return False

        charHash = {}
        for ch in s:
            # if ch in charHash.keys():
            if charHash.get(ch):
                charHash[ch] += 1
            else:
                charHash[ch] = 1
        
        for ch in t:
            if not charHash.get(ch):
                return False
            else: 
                charHash[ch] -= 1
        return True    

File: solutions/Valid Anagram/Valid Anagram.ts
function isAnagram(s: string, t: string): boolean {
    if (s.length !== t.length) return false; // can't be anagrams if not same length

    const sFreq: number[] = new Array(26).fill(0);
    for (let i=0; i<s.length; i++) {
        sFreq[ s[i].charCodeAt(0) - "a".charCodeAt(0) ]++;
    }

    for (let i=0; i<t.length; i++) {
        sFreq[ t[i].charCodeAt(0) - "a".charCodeAt(0) ]--;
    }

    for (let i=0; i<26; i++) {
        if ( sFreq[i] > 0 ) return false;
    }

    return true;
};